src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1135,7 +1136,8 @@ protected virtual TypeDefinition MarkType (TypeReference reference)
 				if (ShouldMarkTypeStaticConstructor (type))
 					MarkStaticConstructor (type);
 
-				MarkMethodsIf (type.Methods, HasSerializationAttribute);
+				if (!_context.IsFeatureExcluded ("deserialization"))
+					MarkMethodsIf (type.Methods, HasSerializationAttribute);
```

Comment:
This marks also serialization methods, if the feature is about deserialization only we should exclude them